pkg/cmd/cli/config/smart_merge.go

```diff
@@ -13,13 +14,75 @@ import (
 	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 
+	osclient "github.com/openshift/origin/pkg/client"
 	"github.com/openshift/origin/pkg/cmd/flagtypes"
 )
 
 var invalidSafeStringSep = regexp.MustCompile(`[.:/]`)
 
+func getClusterNickname(clientCfg *client.Config) (string, error) {
+	serverAddr := flagtypes.Addr{}
```

Comment:
is this for well-formedness and default port addition? if so, this might be a more straightforward way:
```
u, err := url.Parse(clientCfg.Host)
if err != nil {
  return "", err
}
addr := netutil.CanonicalAddr(u)